### PR TITLE
Fixing various issues I found when running through the tutorial

### LINKED ISSUE #59: Todo / view / step 11 confusion
theres  a bit of a wrinkle on todo app doc, view page, section 11. you show the changed view() and view_filters(), but not the changed view_footer(). Section 12 then immediately changes view_footer again, so it sorts itself out, but in theory I can't get the code to compile between sections 11 and 12 (without knowing what I'm doing and fixing view_footer() ).

### LINKED ISSUE #61: todo / update steps 1 and 2 code is mixed up
step 1 changes update for RemoveTodo but view for ToggleTodo. Step 2 does the opposite.

### LINKED ISSUE #62: todo / update - suggest changing order of steps.
The first item we implement is toggle todo, then remove todo. But we don't actually get to create todo until step 4. I suggest moving steps 3 (new todo title updated) and 4 (create todo) up to the top so that when we get to the toggle and remove, I can immediately see them have an effect.

Either that or earlier on in the tutorial, have the default model start with some pre-existing todos. (see also #60 )

### LINKED ISSUE #66: todo / routing / first code block isn't showing the right bits
Code is trying to show filter getting into view_main, but shows view_footer instead:
```rust
fn view(model: &Model) -> Vec<Node<Msg>> {
    ...
            view_footer(&model.todos, model.filter),
...

fn view_main(todos: &BTreeMap<Ulid, Todo>, selected_todo: Option<&SelectedTodo>, filter: Filter) -> Node<Msg> {
    ...
        view_todo_list(todos, selected_todo, filter),
...
```

(I'll probably fix up these issues when I get to the end of the tutorial)

### PR DESCRIPTION
**Pull request by tocklime**
Fixes #59, Fixes #61, Fixes #62 and Fixes #66

### CODE REVIEW

**Comment by tocklime** [comment]
Oops, the indentation thing from #59 is also fixed in #65, but I guess git can figure it out.

File: crate/guides/0.7.0/todomvc_routing.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ Let's implement proper filter handling in our `view` functions. (We could do it
 ```rust
 fn view(model: &Model) -> Vec<Node<Msg>> {
     ...
-            view_footer(&model.todos, model.filter),
+            view_main(&model.todos,model.selected_todo.as_ref(), model.filter),
```
Comment (MartinKavik) [inline crate/guides/0.7.0/todomvc_routing.md:14]: I'm surprised that formatter didn't add a space between `,` and `model.se..` => `&model.todos,model.selected_todo.as_ref()`.

Comment (MartinKavik) [reply crate/guides/0.7.0/todomvc_routing.md:14]: Please just add a space and maybe try to find out why the formatter hasn't fixed it. And I'll merge it.

**Comment by MartinKavik** [comment]
Could you also fix `BtreeMap` - https://github.com/seed-rs/seed-rs.org/pull/65/commits/f7f4c6b4ff0c91e2cfe4153d6887168e96649342 - I thought you've already fixed it in this PR, but I was wrong.

**Comment by MartinKavik** [comment]
Aaaand while you are working on it.. there are other two simple changes... https://github.com/seed-rs/seed-rs.org/pull/39 and https://github.com/seed-rs/seed-rs.org/pull/38 😄 Thank you.

File: crate/guides/0.7.0/todomvc_routing.md (hunk 1 above)
Comment (tocklime) [reply crate/guides/0.7.0/todomvc_routing.md:14]: Formatter doesn't do markdown files, right? I'll put the space in.

Comment (MartinKavik) [reply crate/guides/0.7.0/todomvc_routing.md]: Yeah you are right, I was thinking about the code in the repo, sorry for confusion. It looks like I copied the code before formatting, my bad.

**Pull request merged**